apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.UnitTests/ClientPoolTests.cs

```diff
@@ -20,86 +32,85 @@ public ClientPoolTests(ITestOutputHelper outputHelper)
             TestLogger.Install();
             Logger.LogLevel = V1.Logging.LogLevel.Debug;
         }
-        private static Mock<ISpannerClientFactory> SetupMockClientFactory(Mock<SpannerClient> mockClient)
+
+        private static MockClientFactory SetupMockClientFactory(Mock<SpannerClient> firstClient = null)
         {
-            var mockClientFactory = new Mock<ISpannerClientFactory>();
-            mockClientFactory.Setup(
-                    x => x.CreateClientAsync(It.IsAny<ServiceEndpoint>(), It.IsAny<ITokenAccess>()))
-                .ReturnsAsync(
-                    () =>
-                    {
-                        //return a unique instance each time.
-                        var result = mockClient.Object;
-                        mockClient = new Mock<SpannerClient>();
-                        return result;
-                    });
-            return mockClientFactory;
+            if (firstClient == null)
+            {
+                firstClient = new Mock<SpannerClient>();
+            }
```

Comment:
If you push this into the constructor of `MockClientFactory`, you can remove `SetupMockClientFactory`